### PR TITLE
refactor(sft): unify SFT loss/metrics API

### PR DESCRIPTION
**Pull request by guozhen1997**
### Description

Refactored FSDP SFT training to use a (loss, step_metrics) return contract from `get_train_model_output`:

1. `FSDPSftWorker`: uses the returned loss directly for backward; aggregates per-micro-batch metrics (including loss) via step_metrics.

2. `FSDPVlmSftWorker` / `FSDPVlaSftWorker`: updated to the new API; VLA worker now uses a single training path for OpenPI, LingBot, and DreamZero.

3. `OpenPi0ForRLActionPrediction.sft_forward`: centralized batch preprocessing, device transfer, and loss reduction; supports both dataloader tuples and dict inputs; added use_action_chunk_loss for DAgger.

4. fsdp_actor_worker / fsdp_dagger_policy_worker: removed duplicate OpenPI preprocessing and loss reduction logic.

### Motivation and Context

Previously, SFT workers mixed backward loss computation, metric logging, and model-specific preprocessing in multiple places (base worker, VLA worker overrides, actor/DAgger workers). This made it harder to add model-specific metrics (e.g. DreamZero dynamics_loss) and led to duplicated OpenPI handling.

This change separates training loss from logging metrics, moves OpenPI-specific logic into the model, and unifies VLA SFT call sites—making the SFT pipeline easier to extend and maintain.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by FxxxxU — APPROVED** [review]

**Review by thereAreDemonsNearby — APPROVED** [review]

File: rlinf/workers/sft/fsdp_sft_worker.py [hunk 1]
```diff
@@ -189,7 +180,6 @@ def run_training(self):
             append_to_dict(
                 metrics,
                 {
-                    "loss": avg_loss,
```
Comment (thereAreDemonsNearby) [inline rlinf/workers/sft/fsdp_sft_worker.py:192]: You deleted this loss item, but the loss in **metrics** is a list, not a scalar

Comment (guozhen1997) [reply rlinf/workers/sft/fsdp_sft_worker.py:192]: loss in metrics is a scalar, it is handled and returned in step_metrics by `get_train_model_output` function in `FSDPVlaSftWorker` and `FSDPVlmSftWorker`

**Review by andylin-hao — APPROVED** [review]
